### DISCUSSION TITLE
Typegen CLI breaking values elsewhere

### DISCUSSION THREAD

**Discussion by Ghosts** [chakra-ui/chakra-ui#9226] [Q&A] [answered]
I want custom theme color tokens available for type checking, I have a `theme.ts` file:

```typescript
import { createSystem, defaultConfig, defineConfig } from '@chakra-ui/react';

const customTheme = defineConfig({
    strictTokens: true,
    globalCss: {
        'html, body': {
            margin: '0px',
            padding: '0px',
            fontFamily: 'Inter, sans-serif',
        },
    },
    theme: {
        tokens: {
            colors: {
                background: { value: '#0d0d0d' },
                calamity050: { value: '#fef2f3' },
                calamity100: { value: '#fee2e4' },
               <...Cut for Brevity...>
                white70: { value: '#ffffffb2' },
                white80: { value: '#ffffffcc' },
                white90: { value: '#ffffffe5' },
                whiteWhite: { value: '#ffffff' },
                defaultThemeBackground: { value: '#181818' },
                defaultThemeText: { value: '#ffffff' },
                defaultThemeAccentColor: { value: '#06aaf1' },
            },
        },
    },
});

export const themeSystem = createSystem(defaultConfig, customTheme);

export default themeSystem;
```

When running `npm exec @chakra-ui/cli typegen app/styles/theme.ts` (or the `npx` equivalent`) the command succeeds, but then after restarting my TS server many values that were valid previously are now marked invalid (except for custom colors):

![image](https://github.com/user-attachments/assets/8394cdd9-1525-4041-975c-acb364725f46)
![image](https://github.com/user-attachments/assets/e2294061-ec42-41d3-8ee0-0ff1349dc55d)

Any idea what could be going on? Appreciate any help!

**Comment by segunadebayo** [comment]
I don't see any issues here. Your config has `strictTokens: true` set, and TS shows errors for hardcoded tokens.

Consider setting `strictTokens: false` and see if it works again.

**Reply by Ghosts** [reply]
Ah I guess I misunderstood the flag, I had thought it would enforce tokens by known names / types but allow for valid CSS as well, for example if I have the token `background` but misspell as `backgroound` there is nothing in place to catch that it's not a known token or valid CSS value... is that a possible check at all? Otherwise the token types from the CLI just help for auto complete and not valid type checking.

**Comment by segunadebayo** [comment, marked as answer]
`strictTokens: true` rejects raw CSS on purpose. That is the flag working.

Turn it off, or use token paths.
